pkg/image/export/main.go

```diff
@@ -344,48 +338,200 @@ echo "Creating ${images} with $(echo ${pulled} | wc -w | tr -d '[:space:]') imag
+for ($i = $args.Length; $i -ge 0; $i--)
+{
+    $arg = $args[$i]
+    switch -regex ($arg)
+    {
+        '^(-i|--images)$' {
+            $images = ($vals | Select-Object -First 1)
+            $vals = $null
+        }
+        '^(-l|--image-list)$' {
+            $image_list = ($vals | Select-Object -First 1)
+            $vals = $null
+        }
+        '^(-r|--registry)$' {
+            $registry = ($vals | Select-Object -First 1)
+            $vals = $null
+        }
+        '^(-o|--os-release-id)$' {
+            $os_release_id = ($vals | Select-Object -First 1)
+            $vals = $null
+        }
+        '^(-h|--help)$' {
+            $help = $true
+            $vals = $null
+        }
+        default {
+            if ($vals) {
+                $vals = ,$arg + $vals
+            } else {
+                $vals = @($arg)
+            }
+        }
+    }
+}
 
-foreach ($item in $content) {
-    docker tag $item $Registry/$item
-    docker push $Registry/$item
+if ($help)
+{
+    usage
+    exit 0
 }
+
+if (-not $registry)
+{
+	echo "Must input registry address !"
+    usage
+    exit 1
+}
+
+if (-not (Test-Path $images))
+{
+	echo "Could not find '$images' tar.gz !"
+    usage
+	exit 1
+}
+
+docker load --input $images
+if (-not $?)
+{
+	echo "Could not load '$images'"
+    exit 1
+}
+
+if (-not (Test-Path $image_list))
+{
+	exit 0
+}
+
+Get-Content -Force -Path $image_list | ForEach-Object {
+    if ($_) {
+        $fullname_image = ('{0}-windows-{1}' -f $_, $os_release_id)
+		echo "Tagging $registry/$fullname_image"
+	
+		switch -regex ($fullname_image)
+		{
+			'.+/.+' {
+				docker tag $fullname_image $registry/$fullname_image
+                if ($?) {
+                    docker push $registry/$fullname_image
+                }
+			}
+			default {
+				docker tag $fullname_image $registry/rancher/$fullname_image
+				if ($?) {
+                    docker push $registry/rancher/$fullname_image
+                }
+			}
+		}
+    }
+}
+
 `
-	windowsSaveScirpt = `#Requires -Version 5.0
-<#
-	.PARAMETER image-list
-	text file with list of images. 1 per line.
-	.PARAMETER images
-	tar.gz generated by docker save.
-#>
-param(
-	[string]${image-list}="rancher-windows-images.txt",
-	[string]$images="rancher-windows-images.tar.gz"
-)
+	windowsSaveScript = `$ErrorActionPreference = 'Stop'
+
+$script_name = $MyInvocation.InvocationName
+$image_list = "rancher-windows-images.txt"
+$images = "rancher-windows-images.tar.gz"
+$os_release_id = $(Get-ItemProperty 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\' | Select-Object -ExpandProperty ReleaseId)
+$help = $false
+
+function usage {
+    echo "USAGE: $script_name [--image-list rancher-windows-images.txt] [--images rancher-windows-images.tar.gz]"
+    echo "  [-l|--image-list path] text file with list of images. 1 per line."
+    echo "  [-i|--images path] tar.gz generated by docker save."
+    echo "  [-o|--os-release-id (1809|1903|...)] release id of OS, could detect automatically."
```

Comment:
`release id of OS, gets detected automatically if not passed."`

Reply:
Once changed, should be done in usage command as well: https://github.com/rancher/rancher/pull/22880/files#diff-6bea58749a943397efd8c4a9b0e75351R350